### ISSUE TITLE
Files deleted after 1h

### ISSUE THREAD

**Issue by zell180** [somenonymous/OshiUpload#14]
Hi, I know that the topic is not directly related to the project but it is the only contact channel I have found.
Why are my files deleted even though I entered the expire = 7200 parameter in the POST request?

**Comment by somenonymous** [comment]
Does this happen on your your self-hosted instance or `oshi.at` ?

**Comment by zell180** [comment]
On oshi.at

**Comment by somenonymous** [comment]
We currently run an external script hourly that lists archives containing executable files and automatically delete them. Since you have mentioned your files are being removed after an hour then they're probably caught by this script. We do this due to a recent malware influx that was happening on our domain that made our domain flagged by every possible blacklist provider, thus this measure had to be introduced. The following extensions trigger automatic removal: `exe`,`apk`,`dll`

**Comment by zell180** [comment]
Yes i confirm inside there is an exe file. Is an rpgmaker game. There is no way to sign this file as trusted?

**Comment by somenonymous** [comment]
Making such exceptions will presumably open possibilities to circumvent our measures by impersonating one of the whitelisted files. Our script also doesn't extract them from the archive, thus hashsum whitelisting isn't an option here either, because it only lists archive filenames.

You still have various options to avoid it:

1. Encrypt filenames in the archive using 7-zip `-mhe` flags, so our script will just skip it: `7z -p -mhe a archive.7z [files]`

2. Use an archive type that isn't one of these: `rar`, `zip`, unencrypted `7z`

**Comment by somenonymous** [comment]
Making such exceptions will presumably open possibilities to circumvent our measures by impersonating one of the whitelisted files. Our script also doesn't extract them from the archive, thus hashsum whitelisting isn't an option here either, because it only lists archive filenames.

You still have various options to avoid it:

1. Encrypt filenames in the archive using 7-zip `-mhe` flags, so our script will just skip it: `7z -p -mhe a archive.7z [files]`

2. Use an archive type that isn't one of these: `rar`, `zip`, unencrypted `7z`

**Comment by zell180** [comment]
Thank you. Probably i can do without It

**Comment by zell180** [comment]
I've deleted exe file but .zip file continue to be deleted. Other type of file that i need to check?

**Comment by somenonymous** [comment]
Yeah, I've just opened a script log and your `dll` also get triggered:

```
game/rpg_rt.exe [ boMy_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ iabu_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ mBdR_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ Nncd_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ Xzbw_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ nxHu_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ nght_patch.145.rpgilmito.zip ]
Files deleted: 1
game/rpg_rt.exe [ hogT_patch.145.rpgilmito.zip ]
game/ultimate_rt_eb.dll [ wNNU_patch.145.rpgilmito.zip ]
game/rpg_rt.exe [ kHjQ_patch.145.rpgilmito.zip ]
Files deleted: 3
game/ultimate_rt_eb.dll [ CFGj_patch.145.rpgilmito.zip ]
Files deleted: 1
pol/ImportXSL.dll [ GjHi_Pro_Swapper.rar ]
game/ultimate_rt_eb.dll [ WRDR_patch.145.rpgilmito.zip ]
Files deleted: 2
```

> The following extensions trigger automatic removal: `exe`,`apk`,`dll`

**Comment by zell180** [comment]
Damn.. so dll and exe with zip file is not compatible.. absolutely no way to accomplish this task?

**Comment by somenonymous** [comment]
There is a way:


> You still have various options to avoid it:
> 
> - Encrypt filenames in the archive using 7-zip -mhe flags, so our script will just skip it: `7z -p -mhe a archive.7z [files]`
> -  Use an archive type that isn't one of these: rar, zip, unencrypted 7z


Or just use a .tar archive.
